containers/go-postgres/README.md

```diff
@@ -0,0 +1,165 @@
+
+## Using the PostgreSQL Database
+You can connect to PostgreSQL from an external tool when using VS Code by updating `.devcontainer/devcontainer.json` as follows:
+
+```json
+"forwardPorts": [ "5432" ]
+```
+
+Once the PostgreSQL container has port forwarding enabled, it will be accessible from the Host machine at `localhost:5432`. The [PostgreSQL Documentation](https://www.postgresql.org/docs/14/index.html) has:
+
+1. [An Installation Guide for PSQL], a CLI tool to work with a PostgreSQL database.
+2. [Tips on populating data](https://www.postgresql.org/docs/14/populate.html) in the database. 
+
+If needed, you can use `postCreateCommand` to run commands after the container is created, by updating `.devcontainer/devcontainer.json` similar to what follows:
+
+```json
+"postCreateCommand": "go version && git --version && node --version"
+```
+
+### Adding another service
+
+You can add other services to your `docker-compose.yml` file [as described in Docker's documentation](https://docs.docker.com/compose/compose-file/#service-configuration-reference). However, if you want anything running in this service to be available in the container on localhost, or want to forward the service locally, be sure to add this line to the service config:
+
+```yaml
+# Runs the service on the same network as the database container, allows "forwardPorts" in devcontainer.json function.
+network_mode: service:db
+```
+
+### Installing GO Dependencies
+
+This definition includes the popular [PostGres Driver Library for Go](github.com/lib/pq). This is the recommended driver for use with Go, as per [GoLang Documentation](https://golangdocs.com/golang-postgresql-example).
+
+If you wish to change this, you may add additional `RUN` commands in the [Go Dockerfile](.devcontainer/Dockerfile). For example:
+
+```yaml
+# This line can be modified to add any needed additional packages
+RUN go get -x <github-link-for-package>
+```
+
+## Testing the definition
+
+This definition includes some test code that will help you verify it is working as expected on your system. Follow these steps:
+
+1. If this is your first time using a development container, please follow the [getting started steps](https://aka.ms/vscode-remote/containers/getting-started) to set up your machine.
+2. Clone this repository.
+3. Start VS Code, press <kbd>F1</kbd>, and select **Remote-Containers: Open Folder in Container...**
+4. Select the `containers/go` folder.
+5. After the folder has opened in the container, press <kbd>F5</kbd> to start the project.
+6. From here the following should print out in the Debug Console:
+   ```
+   Connected!
+   Sending Query to Database
+   One database in this cluster is: postgres
+   ```
+7. From here, you can add breakpoints or edit the contents of the `test-project` folder to do further testing.
+8. The Application can also be tested by running `./test.sh` from the Terminal. This will test the `hello.go` application.
+
+
+### Debugging Security
```

Comment:
hi @KanishkT123, I think this next section is the 'wrong' PR -- I'll remove it from the GO PR and put it in the C++ PR :)

Reply:
Hey Dan! Apologies, but that is actually in the right section. All the Go Debuggers that are commonly used are written in C++, so the Go debuggers need the same security permissions.

It should _also_ be in the C++ PR, but it definitely belongs here too. It's why the same security permissions are in both PRs.